lasagne/layers/helper.py

```diff
@@ -148,11 +148,15 @@ def get_output(layer_or_layers, inputs=None, **kwargs):
     both, while the latter will use two different dropout masks.
     """
     from .input import InputLayer
-    from .base import MergeLayer
+    from .base import MergeLayer, Layer
     # track accepted kwargs used by get_output_for
     accepted_kwargs = {'deterministic'}
     # obtain topological ordering of all layers the output layer(s) depend on
     treat_as_input = inputs.keys() if isinstance(inputs, dict) else []
+    for input_key in treat_as_input:
+        if not isinstance(input_key, Layer):
+            raise KeyError("The inputs dictionary keys must be"
+                           " lasagne layers not %s." % type(input_key))
```

Comment:
That's a good place from an execution standpoint, but not so much for reading the code, since it doesn't match the comment. I'd suggest to move it to the beginning of the function, with a comment saying that it checks whether the input directory is valid, even if this means you'll need an additional `if isinstance(inputs, dict)` in front. Also `KeyError` is wrong, that is reserved for missing dictionary keys. It should be a `TypeError` instead. (See https://docs.python.org/2/library/exceptions.html.)